main.go

```diff
@@ -88,17 +89,23 @@ func main() {
 		comparator = wrapReverse(comparator)
 	}
 
-	for i := 0; i < len(dataSplit); i++ {
-		j := i
-		for j > 0 && comparator(dataSplit[j], dataSplit[j-1]) < 0 {
-			dataSplit[j], dataSplit[j-1] = dataSplit[j-1], dataSplit[j]
-			j -= 1
+	if *quickSortParameter {
+		quickSort(dataSplit)
+	} else {
+		for i := 0; i < len(dataSplit); i++ {
```

Comment:
Either all or none shall be outsourced into functions.

Reply:
Am I wrong with putting my sort algo into a function?

Reply:
No, you're wrong with putting some in functions and some not.

Reply:
I did it like that but i guess I misunderstood the exercise.
```
if *quickSortParameter {
		quickSort(dataSplit)
	} else {
		bubbleSort(dataSplit, comparator)
	}

```

```
func bubbleSort(input [][]byte, comp func(a, b []byte) int) [][]byte {
	for i := 0; i < len(input); i++ {
		j := i
		for j > 0 && comp(input[j], input[j-1]) < 0 {
			input[j], input[j-1] = input[j-1], input[j]
			j -= 1
		}
	}
	return input
}
```